proposals/2024-09-25_metadata-labels.md

```diff
@@ -0,0 +1,287 @@
+* Downstream users might be surprised by the new labels e.g. in Cortex, Thanos, Mimir, vendors.
+
+## Potential Future Extensions
+
+Potential extensions, likely requiring dedicated proposals.
+
+### Prometheus UI Changes
+
+When displaying a metric's labels in the table or in the graph views, the UI will hide labels starting with `__` (double underscore) by default, similar to the current handling of `__name__`. A "Show System Labels" check-box might be added, which shows hidden labels when checked.
+
+This is scoped down from the initial implementation due to complexity of the special feature flags in UI and the fact that majority of PromQL users in general might not use Prometheus UI, so we have to educate users on new labels, no matter what we do here.
+
+### More strict unit and type value definition
+
+The current plan delegates the unit and type label definitions to exposition and ingestion formats. Generally, it's not feasible to expect all of the backends to be compatible with all of the
+different types and units. For example for unit [OM unit is free-form string with the bias towards base units](https://prometheus.io/docs/specs/om/open_metrics_spec/#units-and-base-units) for OTLP semantic conventions it's [the UCUM](https://unitsofmeasure.org/ucum) standard. For types OpenMetrics define e.g. stateset, which neither Prometheus, or OTLP natively supports. OTLP defines `UpDownCounter` which does not natively exist in Prometheus or OpenMetrics.
+
+One could try to define standard translations or required subset of supported types in PromQL e.g. [the lowercase OpenMetrics types](https://github.com/prometheus/prometheus/blob/2aaafae36fc0ba53b3a56643f6d6784c3d67002a/model/textparse/openmetricsparse.go#L464). This is essential if we want to have robust type or unit aware functions and operations in PromQL one day. Also, it's critical for the proposal of noticing mixed types being passed through the PromQL engine or auto-converting units.
+
+One alternative is to say OpenMetrics types and units and everything else, before going to PromQL should be translated to OpenMetrics defined types and units. This is a bit limited, because Prometheus does not have native support to stateset and info metrics. We also plan to add delta type, which only exists in OTLP. For units, generally is not strictly defined in OpenMetrics, and [OTLP UCUM](https://unitsofmeasure.org/ucum) generally offers more functionality (e.g. standard way of representing concrete amount of batches of units e.g. 100 seconds).
+
+We propose to raise this problem in the Prometheus ecosystem and tackle it in different proposal. For now, Prometheus will use OpenMetrics type normalization and no unit definition.
```

Comment:
```suggestion
We propose to raise this problem in the Prometheus ecosystem and tackle it in a different proposal. For now, Prometheus will use OpenMetrics type normalization and no unit definition.
```

Reply:
Thanks for grammar fixes, but does this proposal makes sense to you? Do you see the challenge with lack of spec for those types and unit values? 🤔

Reply:
or am I overreacting?